js/views/map.js

```diff
@@ -22,18 +22,41 @@ DeveloperPlayground.MapView = Backbone.View.extend({
         this.stopListening();
     },
 
-    
+
+    hasWebGL: function() {
+        //detect webgl is available on the browser
+        var canvas = document.createElement('canvas'),
+            gl,
+            glExperimental = false;
+
+
+        try { gl = canvas.getContext("webgl"); }
+        catch (x) { gl = null; }
+
+        if(gl) { return true; }
+        else { return false; }
+    },
+
     render: function() {
         this.markerclustergroup = new L.MarkerClusterGroup({showCoverageOnHover: false});
+
         this.map = L.map('map-view',{
             scrollWheelZoom: false
         }).setView([37.749, -122.443], 11);
-        // run setMapview inside of setview
-        L.tileLayer('https://{s}.tiles.mapbox.com/v3/randyme.li1lhlf0/{z}/{x}/{y}.png', {
-            maxZoom: 18,
-            attribution: 'Map data &copy; <a target="_blank" href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a target="_blank" href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a target="_blank" href="http://mapbox.com">Mapbox</a>',
-        })
-        .addTo(this.map);
+
+        var layer;
+        if(this.hasWebGL()) {
+            var layer = Tangram.leafletLayer({
+                 scene: 'https://cdn.rawgit.com/tangrams/multiverse/gh-pages/styles/tangram-toner.yaml',
```

Comment:
Ok, to be clear, Refill is the same style as tangram-toner, but more
complete (tangram-toner was the original name).


On Thursday, October 8, 2015, Hanbyul Jo <notifications@github.com> wrote:

> In js/views/map.js
> <https://github.com/transitland/playground/pull/110#discussion_r41564923>:
>
> >          this.map = L.map('map-view',{
> >              scrollWheelZoom: false
> >          }).setView([37.749, -122.443], 11);
> > -        // run setMapview inside of setview
> > -        L.tileLayer('https://{s}.tiles.mapbox.com/v3/randyme.li1lhlf0/{z}/{x}/{y}.png <http://tiles.mapbox.com/v3/randyme.li1lhlf0/%7Bz%7D/%7Bx%7D/%7By%7D.png>', {
> > -            maxZoom: 18,
> > -            attribution: 'Map data &copy; <a target="_blank" href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a target="_blank" href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a target="_blank" href="http://mapbox.com">Mapbox</a>',
> > -        })
> > -        .addTo(this.map);
> > +
> > +        var layer;
> > +        if(this.hasWebGL()) {
> > +            var layer = Tangram.leafletLayer({
> > +                 scene: 'https://cdn.rawgit.com/tangrams/multiverse/gh-pages/styles/tangram-toner.yaml',
>
> I tried putting refill first, but it didn't look best. (attaching
> screenshot)
> Waiting for Cartography team to get base map ready.
>
>    - Playground will probably stick with raster tiles for now [image:
>    screen shot 2015-10-08 at 4 30 19 pm]
>    <https://cloud.githubusercontent.com/assets/4583806/10378969/0ea48c8c-6dda-11e5-8975-b7d2b9614fa6.png>
>    about base map
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/transitland/playground/pull/110/files#r41564923>.
>